### PR TITLE
migrate SogouNews to datapipes.

### PR DESCRIPTION
**Pull request by erip**
- [x] Migrate dataset in `torchtext/datasets/`

Reference Issue: #1494

### CODE REVIEW

**Comment by erip** [comment]
Ready for review too, @Nayef211 @parmeet @abhinavarora

File: torchtext/datasets/sogounews.py [hunk 1]
```diff
@@ -35,10 +38,17 @@
...
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL])
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
+    )
+    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_dp = FileOpener(cache_dp, mode="b")
+    extracted_files = cache_dp.read_from_tar()
+    filter_extracted_files = extracted_files.filter(lambda x: split + ".csv" in x[0])
```
Comment (Nayef211) [inline torchtext/datasets/sogounews.py:53]: NIT: Let's be consistent with how we're doing the check for the `split` file existing in the folder.
We could do `lambda x: _EXTRACTED_FILES[split] in x[0]` which is what you did in [this PR](https://github.com/pytorch/text/pull/1499/files#diff-61c4706569928ac5352feafe9003545f246296ecc93c10e20120747f3707ccbdR57)

**Review by Nayef211 — COMMENTED** [review]
Let's merge in the latest changes from master to fix the broken `build_docs` job

**Review by Nayef211 — APPROVED** [review]
LGTM!

**Pull request merged**